### PR TITLE
Updating for v5

### LINKED ISSUE #118: Updating for v5
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

👋 I'm Matteo, and would like to work on this.
Coming from Fastify issue #5116 (https://github.com/fastify/fastify/issues/5116)

From what I've understood, I need to:

- Update `tap` package
- Update the workflow to use the version 4.1.0 of plugins-ci
- Drop support to node v14 (there is a specific case in `index.js` for node v14 (line 41)

If there's anything I'm missing, please do let me know

### PR DESCRIPTION
**Pull request by Gesma94**
Closes #118 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

**CC** @simoneb

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -59,7 +52,7 @@ async function fastifyRoutesStats (fastify, opts) {
 
   fastify.addHook('onSend', function (request, reply, _, next) {
     if (request[decoratorName]) {
-      const routeId = reply.context.config.statsId || request.raw.url
+      const routeId = request.routeOptions.config.statsId || request.raw.url
```
Comment (Gesma94) [inline index.js:55]: `reply.context` will be deprecated in Fastify v5, thus I used the new suggested `request.routeOptions.config`

**Comment by gurgunday** [comment]
Also `istanbul` no longer does anything as tap uses `c8` now

You can remove them or replace with `/* c8 ignore next */`

**Comment by Gesma94** [comment]
> Also `istanbul` no longer does anything as tap uses `c8` now
> 
> You should remove them or replace with `/* c8 ignore next */`

Sorry, I missed that. I also forgot to push the workflow file, so the actions were failing: I've updated the `ci.yml` file

**Review by gurgunday — APPROVED** [review]

**Review by Fdawgs — APPROVED** [review]

**Comment by gurgunday** [comment]
Thanks!

**Pull request merged**